### PR TITLE
Front upgrade admin dashboard

### PR DESCRIPTION
**Pull request by Ronfflex**
**This is only to make a sort of preview of how the admin dashboard could look like. Most components might change a lot during the integration with the backend.**

image

Updating dashboard front with different categories and components.

### CODE REVIEW

**Comment by Ronfflex** [comment]
J'ai pas forcément typé proprement mes composants mais j'ai pas envie de perdre trop de temps dessus. Je propose de le faire uniquement vers la fin si on a le temps clean le code.

@MTthoas Au niveau du merge conflict, est-ce que je suis censer garder les fichiers Combobox ?

File: next-app/src/app/administration/analytics/page.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <main className="flex flex-1 flex-col gap-4 p-4 md:gap-8 md:p-6">
+      <NewUsersGraph />
+      <TotalLiquidityGraph />
+      <TotalTradingVolumeGraph />
+    </main>
+  </>
+  );
+};
+
+function SearchIcon(props: React.JSX.IntrinsicAttributes & React.SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/app/administration/analytics/page.tsx:38]: I think one good idea is to centralize icons / svg / png in one icons folder

File: next-app/src/app/administration/contracts/page.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+      </header>
+      <main className="flex flex-1 flex-col gap-4 p-4 md:gap-8 md:p-6">
+        <Contracts />
+      </main>
+    </>
+  );
+};
+
+function SearchIcon(props: React.JSX.IntrinsicAttributes & React.SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/app/administration/contracts/page.tsx:33]: Indeed, you're using the same Svg

File: next-app/src/app/administration/histories/page.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        )}
+        <TransactionsTable
+          transactions={transactions.data ? transactions.data?.data : []}
+        />
+      </main>
+    </>
+  );
+};
+
+function SearchIcon(props: React.JSX.IntrinsicAttributes & React.SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/app/administration/histories/page.tsx:63]: Again

File: next-app/src/app/administration/platform/page.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        <div className="grid grid-cols-1 gap-4 md:grid-cols-1 lg:grid-cols-2">
+          <PlatformFee />
+          <BanUserAddress />
+        </div>
+      </main>
+    </>
+  );
+};
+
+function SearchIcon(props: React.JSX.IntrinsicAttributes & React.SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/app/administration/platform/page.tsx:37]: again

File: next-app/src/app/administration/tokens/page.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+      </header>
+      <main className="flex flex-1 flex-col gap-4 p-4 md:gap-8 md:p-6">
+        <Airdrop />
+      </main>
+    </>
+  );
+};
+
+function SearchIcon(props: React.JSX.IntrinsicAttributes & React.SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/app/administration/tokens/page.tsx:33]: agaib

File: next-app/src/components/administration/analytics/NewUsersGraph.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  CardTitle,
+  CardDescription,
+  CardContent,
+} from "@/components/ui/card";
+import { ResponsiveLine } from "@nivo/line";
+import { JSX, ClassAttributes, HTMLAttributes } from "react";
+
+const NewUsersGraph = () => {
+  return (
+    <div>
```
Comment (MTthoas) [inline next-app/src/components/administration/analytics/NewUsersGraph.tsx:15]: delete this div

File: next-app/src/components/administration/analytics/NewUsersGraph.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        </CardHeader>
+        <CardContent>
+          <LineChart className="aspect-[9/4]" />
+        </CardContent>
+      </Card>
+    </div>
+  );
+};
+
+function LineChart(props: JSX.IntrinsicAttributes & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement>) {
```
Comment (MTthoas) [inline next-app/src/components/administration/analytics/NewUsersGraph.tsx:31]: You're duplicating the same code

File: next-app/src/components/administration/analytics/TotalTradingVolumeGraph.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  CardTitle,
+  CardDescription,
+  CardContent,
+} from "@/components/ui/card";
+import { ResponsiveLine } from "@nivo/line";
+import { JSX, ClassAttributes, HTMLAttributes } from "react";
+
+const TotalTradingVolumeGraph = () => {
+  return (
+    <div>
```
Comment (MTthoas) [inline next-app/src/components/administration/analytics/TotalTradingVolumeGraph.tsx:15]: div to delete

File: next-app/src/components/administration/analytics/TotalLiquidityGraph.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  CardTitle,
+  CardDescription,
+  CardContent,
+} from "@/components/ui/card";
+import { ResponsiveBar } from "@nivo/bar";
+import { JSX, ClassAttributes, HTMLAttributes } from "react";
+
+const TotalLiquidityGraph = () => {
+  return (
+    <div>
```
Comment (MTthoas) [inline next-app/src/components/administration/analytics/TotalLiquidityGraph.tsx:15]: div to delete

File: next-app/src/components/administration/analytics/TotalTradingVolumeGraph.tsx [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+        </CardHeader>
+        <CardContent>
+          <TimeseriesChart className="aspect-[9/4]" />
+        </CardContent>
+      </Card>
+    </div>
+  );
+};
+
+function TimeseriesChart(props: JSX.IntrinsicAttributes & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement>) {
```
Comment (MTthoas) [inline next-app/src/components/administration/analytics/TotalTradingVolumeGraph.tsx:31]: Put charts in one folder

File: next-app/src/components/administration/contracts/Contracts.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  Card,
+  CardHeader,
+  CardTitle,
+  CardDescription,
+  CardContent,
+} from "@/components/ui/card";
+
+const Contracts = () => {
+  return (
+    <div>
```
Comment (MTthoas) [inline next-app/src/components/administration/contracts/Contracts.tsx:15]: div

File: next-app/src/components/administration/contracts/Contracts.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      <Card>
+        <CardHeader>
+          <CardTitle>Contract Management</CardTitle>
+          <CardDescription>
+            View and manage the smart contract balance.
+          </CardDescription>
+        </CardHeader>
+        <CardContent>
+          <div className="grid gap-4">
+            <div>
```
Comment (MTthoas) [inline next-app/src/components/administration/contracts/Contracts.tsx:25]: div

File: next-app/src/components/administration/platform/PlatformFee.tsx [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+              </CardContent>
+            </Card>
+          </div>
+        )}
+      </Card>
+    </div>
+  );
+};
+
+function CircleCheckIcon(props: JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>) {
```
Comment (MTthoas) [inline next-app/src/components/administration/platform/PlatformFee.tsx:73]: Icon to folder

File: next-app/src/components/administration/tokens/Airdrop.tsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { JSX, SVGProps, useState } from "react";
+
+const Airdrop = () => {
+  const [address, setAddress] = useState("");
+  const [error, setError] = useState("");
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const handleSaveChanges = () => {
+    const addressRegex = /^0x[a-fA-F0-9]{40}$/g;
```
Comment (MTthoas) [inline next-app/src/components/administration/tokens/Airdrop.tsx:19]: duplicated const

File: next-app/src/components/administration/tokens/Airdrop.tsx [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { JSX, SVGProps, useState } from "react";
+
+const Airdrop = () => {
+  const [address, setAddress] = useState("");
+  const [error, setError] = useState("");
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const handleSaveChanges = () => {
```
Comment (MTthoas) [inline next-app/src/components/administration/tokens/Airdrop.tsx:18]: Duplicated hook

File: next-app/src/components/administration/tokens/Airdrop.tsx [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+              </CardContent>
+            </Card>
+          </div>
+        )}
+      </Card>
+    </div>
+  );
+};
+
+function CircleCheckIcon(
```
Comment (MTthoas) [inline next-app/src/components/administration/tokens/Airdrop.tsx:89]: Icon to folder

File: next-app/src/components/administration/tokens/Airdrop.tsx (hunk 14 above)
Comment (Ronfflex) [reply next-app/src/components/administration/tokens/Airdrop.tsx:19]: I don't understand or can't see it.

File: next-app/src/components/administration/tokens/Airdrop.tsx (hunk 15 above)
Comment (Ronfflex) [reply next-app/src/components/administration/tokens/Airdrop.tsx:18]: Duplicated is the good word ?? I don't get it

**Pull request merged**